// trip/trip-manager.js
var fs = require('fs');
var path = require('path');

exports.getTrip = function(tripId) {
  console.log('download trip information for ' + tripId);

  var filePath = getResourceURL(findTripPath(tripId));

  console.log('Resolved file path ' + filePath);
  var tripStat = fs.statSync(filePath);

  var tripStream = fs.createReadStream(filePath);  

  var tripInfo = {
    stream : tripStream,
    stat : tripStat,
    contentInfo : 'application/xml'
  };

  return tripInfo;
};

exports.getInstruction = function(instructionId) {
  var filePath = getResourceURL(findInstructionPath(instructionId));
  var instructionStat = fs.statSync(filePath);

  var instructionStream = fs.createReadStream(filePath);
  var instructionInfo = {
    stream : instructionStream,
    stat : instructionStat,
    contentInfo : 'audio/mpeg'
  };

  return instructionInfo;
}

function getResourceURL(resourceFullName) {
    return path.join('res/tripDb/', resourceFullName);
}

function findTripPath(tripId) {
  return tripId + '.xml';
}

function findInstructionPath(instructionId) {
  return instructionId + '.mp3';
}
